### ISSUE TITLE
Some questions about "user settings"  and "emmet" & "snippet" 

### ISSUE THREAD

**Issue by Meli55a** [Microsoft/vscode#13584]
- VSCode Version: 1.6.0
- OS Version: windows 10 x64

> I am a Chinese, My English is not very good, these words are written using translation software, Excuse me.
> 
> I like VScode very much，I used it from V0.9 onwards, from the first day I know it, together with a strong recommendation to a friend, although there are still some problems, but I believe it will be getting better and better!

There are a few problems have always wanted to say, but my English is not good, today I have the courage to say it, hoping to improve.
1. user settings：

When we set some options, except the "true" or  "false" option, there is no hint at the value of the comment line, like:

``` json
    // 控制编辑器是否应呈现空白字符
    "editor.renderWhitespace": "none",
    // 控制是否将代码段与其他建议一起显示以及它们的排序方式。
    "editor.snippetSuggestions": "bottom",

--------------------I wish it could be like the following------------
    // 控制编辑器是否应呈现空白字符, valid values: ["none","boundary","all"]
    "editor.renderWhitespace": "none",
    // 控制是否将代码段与其他建议一起显示以及它们的排序方式。valid values: ["top","bottom","inline","none"]
    "editor.snippetSuggestions": "bottom",

This will be very convenient for us to set up
```
1.  "emmet" & "snippet" 

In VS code, when we enter the PHP built-in function, like **"echo"**, when we enter "echo", It will appear smart tips, but when we press the `tab` button, it just shows `echo` .

In Sublime Text, when we enter `echo` and press the `tab`, it is `'echo "string";'` , and the cursor on the `string`, it greatly saving times.

``` php
// enter `echo` and press tab:
------------In VS code------------
echo

------------In sublime text------------
echo "string";
```

**Comment by ramya-rao-a** [comment]
@Meli55a

For your first question, we do have intellisense that show you the options for the settings.
You can hit `Ctrl+Space` from inside the double quotes to trigger it.

![image](https://cloud.githubusercontent.com/assets/16890566/20152749/618d77c8-a674-11e6-85ac-6f7d31cbdfea.png)





 For you second question, have you tried creating custom user snippets?

**Comment by Meli55a** [comment]
Thank you!, he first problem has been solved. For the second question, i know hao to creating custom user snippets, I hope vs code  native support

**Comment by ramya-rao-a** [comment]
@alexandrudima @aeschli Do we take snippet requests for languages?

**Comment by aeschli** [comment]
@ramya-rao-a We don't want to add snippets to our built-in extension. Snippets should be in extensions from the marketplace as this gives users more flexibility to control what snippets are available.
Some users don't like some our snippets and there's currently no way of hiding or replacing a single snippet.

**Comment by ramya-rao-a** [comment]
Thanks @aeschli.
@Meli55a, you must be using some PHP related extension, you can get in touch with the extension owner to include this snippet

**Comment by Meli55a** [comment]
Thanks @ramya-rao-a @aeschli , I tried some PHP extensions. but they are not ideal, i hope the future will be better, I wish you a pleasant job. @ramya-rao-a your smile is pleasan! Thanks again.
